### PR TITLE
Folder-based save picker for iOS

### PR DESCRIPTION
**Pull request by MartinZikmund**
Closes https://github.com/unoplatform/uno/issues/5229

Sample for folder-based save picker on iOS.

### CODE REVIEW

File: UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/App.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        /// Invoked when the application is launched normally by the end user.  Other entry points
+        /// will be used such as when the application is launched to open a specific file.
+        /// </summary>
+        /// <param name="e">Details about the launch request and process.</param>
+        protected override void OnLaunched(LaunchActivatedEventArgs e)
+        {
+#if DEBUG
+            if (System.Diagnostics.Debugger.IsAttached)
+            {
+                // this.DebugSettings.EnableFrameRateCounter = true;
```
Comment (ajpinedam) [inline UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/App.xaml.cs:55]: will this be removed?

File: UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/App.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            if (rootFrame == null)
+            {
+                // Create a Frame to act as the navigation context and navigate to the first page
+                rootFrame = new Frame();
+
+                rootFrame.NavigationFailed += OnNavigationFailed;
+
+                if (e.PreviousExecutionState == ApplicationExecutionState.Terminated)
+                {
+                    //TODO: Load state from previously suspended application
```
Comment (ajpinedam) [inline UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/App.xaml.cs:79]: Will this be addressed in the near future? Otherwise makes sense to comment out the whole `if` block

File: UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/Extensions/FolderSavePickerDialog.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+            {
+                return;
+            }
+
+            var currentText = sender.Text;
+            var matchingFileNames = _existingFileNames
+                .Where(n => n != null && n.Contains(currentText, StringComparison.CurrentCultureIgnoreCase) &&
+                    (!Path.HasExtension(n) ||
+                     currentFileTypeChoice.Extensions.Contains(Path.GetExtension(n))))
+                .Take(5)
```
Comment (ajpinedam) [inline UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/Extensions/FolderSavePickerDialog.xaml.cs:163]: a constant with this value?

File: UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/MainPage.xaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    x:Class="FileSavePickeriOS.MainPage"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:local="using:FileSavePickeriOS"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    mc:Ignorable="d">
+
+    <Grid Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+        <Button Click="SaveFile_Click" HorizontalAlignment="Center" VerticalAlignment="Center">Save as...</Button>
```
Comment (ajpinedam) [inline UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/MainPage.xaml:11]: Resource-it so it could be Localized?

File: UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/Strings/en/Resources.resw [hunk 5]
```diff
@@ -0,0 +1,168 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (ajpinedam) [inline UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/Strings/en/Resources.resw:1]: Quieres la traduccion en Español?  :)

**Review by ajpinedam — APPROVED** [review]

File: UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/App.xaml.cs (hunk 1 above)
Comment (MartinZikmund) [reply UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/App.xaml.cs:55]: This is coming from default Uno template, but it is not relevant to the sample, the important code is the extension itself 👍

File: UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/App.xaml.cs (hunk 2 above)
Comment (MartinZikmund) [reply UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/App.xaml.cs:79]: This is coming from default Uno template, but it is not relevant to the sample, the important code is the extension itself 👍

File: UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/MainPage.xaml (hunk 4 above)
Comment (MartinZikmund) [reply UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/MainPage.xaml:11]: This button is not part of the extension yet, it only triggers the `FileSavePicker`, so I suppose it does not need to be localized specifically

File: UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/Strings/en/Resources.resw (hunk 5 above)
Comment (MartinZikmund) [reply UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/Strings/en/Resources.resw:1]: Definitely :-) ! If you can provide me the translated strings, I will add them to the PR :-) (or you can commit to this PR directly if you have access :-) )

File: UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/MainPage.xaml (hunk 4 above)
Comment (davidjohnoliver) [reply UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/MainPage.xaml:11]: it's a sample not a production app - localization is overkill here

**Review by davidjohnoliver — APPROVED** [review]

File: UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/App.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        /// <summary>
+        /// Initializes the singleton application object.  This is the first line of authored code
+        /// executed, and as such is the logical equivalent of main() or WinMain().
+        /// </summary>
+        public App()
+        {
+            // Register the IFileSavePickerExtension for iOS.
+#if __IOS__
+            ApiExtensibility.Register(typeof(Uno.Extensions.Storage.Pickers.IFileSavePickerExtension), (picker) => new FolderSavePickerExtension(picker));
+#endif
```
Comment (carldebilly) [inline UI/FileSavePickeriOS/FileSavePickeriOS/FileSavePickeriOS.Shared/App.xaml.cs:37]: Why is this required for a sample? You should document the reason in the code.

**Review by carldebilly — APPROVED** [review]
